modules/scan/class-admin-bar-notice.php

```diff
@@ -0,0 +1,200 @@
+			'multiple'           => sprintf( esc_html__( '%s Threats found', 'jetpack' ), $this->get_icon() ),
+		);
+		wp_localize_script( self::SCRIPT_NAME, 'Jetpack_Scan', $script_data );
+	}
+
+	/**
+	 * Adds the inline styles if they are needed.
+	 */
+	public function add_inline_styles() {
+		// We know there are no threats so lets not include any css.
+		if ( false === $this->has_threats() ) {
+			return;
+		}
+
+		// We might be showing the threats in the admin bar lets make sure that they look great!
+		$hide_wording_on_mobile = '@media screen and (max-width: 959px ) { #wpadminbar #wp-admin-bar-jetpack-scan-notice { width:32px; } #wpadminbar #wp-admin-bar-jetpack-scan-notice a { color: transparent!important; }';
+		$style                  = '#wp-admin-bar-jetpack-scan-notice svg { float:left; margin-top: 4px; margin-right: 6px; width: 18px; height: 22px; }' . $hide_wording_on_mobile;
+		if ( is_rtl() ) {
+			$style = '#wp-admin-bar-jetpack-scan-notice svg { float:right; margin-top: 4px; margin-left: 6px; width: 18px; height: 22px; }' . $hide_wording_on_mobile;
+		}
+		wp_add_inline_style( 'admin-bar', $style );
+	}
+
+	/**
+	 * Add the link to the admin bar.
+	 *
+	 * @param WP_Admin_Bar $wp_admin_bar WP Admin Bar class object.
+	 */
+	public function add_threats_to_toolbar( $wp_admin_bar ) {
+		if ( ! $this->should_try_to_display_notice() ) {
+			return;
+		}
+
+		$has_threats = $this->has_threats();
+		if ( false === $has_threats ) {
+			return;
+		}
+
+		$node = array(
+			'id'     => 'jetpack-scan-notice',
+			'title'  => '',
+			'parent' => 'top-secondary',
+			'meta'   => array(
+				'title' => esc_attr__( 'Visit your scan dashboard', 'jetpack' ),
+				'class' => 'error',
+			),
+		);
+
+		if ( $has_threats ) {
+			$node['href']            = esc_url( Redirect::get_url( 'calypso-scanner' ) );
+			$node['meta']['onclick'] = 'window.open( this.href ); return false;';
+			$node['title']           = sprintf(
+				esc_html(
+				/* translators: %s is the alert icon */
+					_n( '%s Threat found', '%s Threats found', $this->get_threat_count(), 'jetpack' )
+				),
+				$this->get_icon()
+			);
+		}
+
+		$wp_admin_bar->add_node( $node );
+	}
+
+	/**
+	 * Returns the shield icon.
+	 *
+	 * @return string
+	 */
+	private function get_icon() {
+		return '<svg width="18" height="22" viewBox="0 0 18 22" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M9 0L0 4V10C0 15.55 3.84 20.74 9 22C14.16 20.74 18 15.55 18 10V4L9 0Z" fill="#D63638"/><path d="M7.99121 6.00894H10.0085V11.9968H7.99121V6.00894Z" fill="#FFF"/><path d="M7.99121 14.014H10.0085V15.9911H7.99121V14.014Z" fill="#FFF"/></svg>';
+	}
+
+	/**
+	 *
+	 * Return Whether boolean cached threats exist or null if the state is unknown.
+	 * * @return boolean or null
+	 */
+	public function has_threats() {
+		$scan_state = get_transient( 'jetpack_scan_state' );
+		if ( empty( $scan_state ) ) {
+			return null;
+		}
+
+		return (bool) isset( $scan_state->threats[0] );
```

Comment:
I missed the inline comments that were removed during the clean-up commit. 1075cb3

I think they're helpful in understanding the intent and reasons for various things (the bool, the early return for multisite, etc).

Reply:
re added the comment